src/plugins/autocompletion_providers/Grep.ts

```diff
@@ -1,8 +1,278 @@
 import {PluginManager} from "../../PluginManager";
-import {longAndShortFlag, mapSuggestions, combine} from "./Common";
+import {longFlag, longAndShortFlag, mapSuggestions, combine, anyFilesSuggestionsProvider,
+    Suggestion, styles, anyFilesSuggestions, directoriesSuggestions} from "./Common";
+import {AutocompletionContext} from "../../Interfaces";
+import {isDirectory} from "../../utils/Common";
 
-const options = combine([
-    mapSuggestions(longAndShortFlag("count"), suggestion => suggestion.withDescription("Only a count of selected lines is written to standard output.")),
+// Grep option suggestions based on linux  man file:
+// http://linux.die.net/man/1/grep
+const baseOptions = combine([
+    // Generic Program Information
+    mapSuggestions(longAndShortFlag("version", "V"), suggestion => suggestion.withDescription(`
```

Comment:
Having a clear pattern it's better to create a data-only array of objects, and then map it to suggestions.